core/src/common.cpp

```diff
@@ -362,6 +364,264 @@ namespace irods::http
+	auto add_alg_from_jwk(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwk<jwt::traits::nlohmann_json>& _jwk, std::string_view _alg) -> void {
+		namespace logging = irods::http::log;
+
+		if (_alg == "RS256") {
+			logging::trace("{}: Detected [RS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get modulus parameter (JWA)
+			auto mod{_jwk.get_jwk_claim("n").as_string()};
+
+			// Get exponent parameter (JWA)
+			auto exp{_jwk.get_jwk_claim("e").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::rs256(jwt::helper::create_public_key_from_rsa_components(mod, exp)));
+		}
+		// For symetric excryption, no parameter 'k' is provided
+		// See OpenID Connect Discovery section 3
+		if (_alg == "HS256") {
+			logging::trace("{}: Detected [HS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Attempt to get key value parameter (JWA)
+			auto key{_jwk.get_jwk_claim("k").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::hs256(key));
+		}
+		if (_alg == "ES256") {
+			logging::trace("{}: Detected [ES256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get curve parameter (JWA)
+			auto crv{_jwk.get_jwk_claim("crv").as_string()};
+
+			// Get x coordinate parameter (JWA)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::es256(jwt::helper::create_public_key_from_ec_components(crv, x, y)));
+		}
+	}
+
+	/// Bleh blah
+	///
+	/// \returns A refernce to the provided jwt::verifier, allowing for chaining.
+	auto add_algorithms_to_verifier(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwks<jwt::traits::nlohmann_json>& _jwks, std::string_view _alg) -> jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& {
+		namespace logging = irods::http::log;
+
+		// Get the JWK the access token was signed with. This is optional.
+		// See RFC 7515 Section 4.1.4
+		try {
+			auto jwk{_jwks.get_jwk(decoded_token.get_key_id())};
+			add_alg_from_jwk(_verifier, jwk, _alg);
+		}
+		catch (const std::runtime_error& e) {
+			// We cannot pick out the specific key used...
+			// Find JWKs that match any of the following:
+			//   - 'alg' (optional) matches exactly
+			//   - 'kty' (required) matches 'alg' family [EC,RSA,oct]
+			//   - 'use' (required if sign & encrypt keys exist) to be 'sig'
+			//
+			// Additionally, symetric algoritms it /seems/ that
+			// "... the octects of the 'client_secret'..." are used for the algorithm...
+			const auto algorithm_family{_alg.substr(0,2)};
+			std::string search_string;
+
+			// RSA family
+			if (algorithm_family == "RS") {
+				search_string = "RSA";
+			}
+			// EC family
+			else if (algorithm_family == "ES") {
```

Comment:
ES is correct?

Reply:
Yep. Going to push up commit with more comments soon.

Reply:
Comment now describes algorithm types, where it gets shorthand names from.